src/ErrorProne.NET.CoreAnalyzers/AsyncAnalyzers/ConfigureAwaitRequiredAnalyzer.cs

```diff
@@ -27,38 +26,56 @@ public ConfigureAwaitRequiredAnalyzer()
         /// <inheritdoc />
         protected override void InitializeCore(AnalysisContext context)
         {
-            context.RegisterSyntaxNodeAction(AnalyzeAwaitExpression, SyntaxKind.AwaitExpression);
+            context.RegisterCompilationStartAction(context =>
+            {
+                var compilation = context.Compilation;
+
+                var configureAwaitConfig = ConfigureAwaitConfiguration.TryGetConfigureAwait(context.Compilation);
+                if (configureAwaitConfig != ConfigureAwait.UseConfigureAwaitFalse)
+                {
+                    return;
+                }
+
+                var taskType = compilation.GetTypeByMetadataName(typeof(Task).FullName);
+                if (taskType is null)
+                {
+                    return;
+                }
+
+                var configureAwaitMethods = taskType.GetMembers("ConfigureAwait").OfType<IMethodSymbol>().ToImmutableArray();
+                if (configureAwaitMethods.IsEmpty)
+                {
+                    return;
+                }
+
+                var yieldAwaitable = compilation.GetTypeByMetadataName(typeof(YieldAwaitable).FullName);
+
+                context.RegisterOperationAction(context => AnalyzeAwaitOperation(context, configureAwaitMethods, yieldAwaitable), OperationKind.Await);
```

Comment:
Switched from syntax to operation.

At the end, `GetOperation` is called anyways.